### PR TITLE
Added Rfop_class & Rfop model

### PR DESCRIPTION
**Pull request by Iva-khar**

### CODE REVIEW

File: ratu/models/ratu_models.py [hunk 1]
```diff
@@ -4,18 +4,21 @@
...
-    name = models.CharField(max_length=20)
+    name = models.CharField(max_length=30, unique=True)
 
 class District(models.Model):
+    EMPTY_FIELD = 'empty field'
     region = models.ForeignKey(Region, on_delete=models.CASCADE)
     name = models.CharField(max_length=100)
 
 class City(models.Model):
+    EMPTY_FIELD = 'empty field'
```
Comment (criticus) [inline ratu/models/ratu_models.py:15]: same here

File: ratu/models/ratu_models.py [hunk 2]
```diff
@@ -4,18 +4,21 @@
...
     region = models.ForeignKey(Region, on_delete=models.CASCADE)
     district = models.ForeignKey(District, on_delete=models.CASCADE)
     name = models.CharField(max_length=100)
 
 class Citydistrict(models.Model):
+    EMPTY_FIELD = 'empty field'
```
Comment (criticus) [inline ratu/models/ratu_models.py:21]: and here

File: ratu/models/rfop_models.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+from django.db import models
+
+# Create your models here.
+class Rfop(models.Model):
+    fullname = models.CharField(max_length=100, null=True)
+    address = models.CharField(max_length=300, null=True)
+    kved = models.CharField(max_length=250, null=True)
+    state = models.CharField(max_length=100, null=True)
```
Comment (criticus) [inline ratu/models/rfop_models.py:10]: this has to be in separate table and foreign key in this table

File: ratu/models/ruo_models.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+# Create your models here.
+class Company(models.Model):
+    name = models.CharField(max_length=500, null=True)
+    short_name = models.CharField(max_length=500, null=True)
+    edrpou = models.CharField(max_length=50, null=True)
+    address = models.CharField(max_length=500, null=True)
+    boss = models.CharField(max_length=250, null=True)
+    kved = models.CharField(max_length=500, null=True)
+    state = models.CharField(max_length=500, null=True)
```
Comment (criticus) [inline ratu/models/ruo_models.py:13]: separate table?

File: ratu/services/main.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
+import codecs
+import io
+from ratu.models.ratu_models import Region, District, City, Citydistrict, Street
+import requests
+import sys
+import xml.etree.ElementTree
+from xml.etree.ElementTree import iterparse, XMLParser, tostring
```
Comment (criticus) [inline ratu/services/main.py:7]: why do you import `xml.etree.ElementTree` twice?

File: ratu/services/main.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+import xmltodict
+import zipfile
+#pip3 freeze > requirements.txt - must run after add or change import modules
```
Comment (criticus) [inline ratu/services/main.py:10]: please remove unneeded comments

File: ratu/services/main.py [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+class Converter:
+
```
Comment (criticus) [inline ratu/services/main.py:13]: remove extra empty lines

File: ratu/services/main.py [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+                
+                i=i+1
+                print(i, ' records\n\n................................................................................................')
+                for key in record:
+                    if type(record[key]) == list:
+                        record[key].clear()
+                    else:
+                        record[key]=''
+                root.clear()
+        print('All the records have rewrited.')
```
Comment (criticus) [inline ratu/services/main.py:75]: have been rewritten

File: ratu/services/ratu_service.py [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+        self.save_to_district_table(record)
+        self.save_to_city_table(record)
+        self.save_to_citydistrict_table(record)
+        self.save_to_street_table(record)
+        print('saved')
+    
+    #writing entry to region table           
+    def save_to_region_table(self, record):
+        if not record['OBL_NAME'] in self.region_list:
+            global region
```
Comment (criticus) [inline ratu/services/ratu_service.py:49]: please don't use globals, replace it with parameter

File: ratu/services/ratu_service.py [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+                )
+    
+    #writing entry to district table    
+    def save_to_district_table(self, record):
+        if record['REGION_NAME']:
+            a=record['REGION_NAME']
+        else:
+            a=District.EMPTY_FIELD
+        if not [region, a] in self.district_list:
+            global district
```
Comment (criticus) [inline ratu/services/ratu_service.py:66]: same here with globals

File: ratu/models/ratu_models.py [hunk 11]
```diff
@@ -4,18 +4,21 @@
 
 # Create your models here.
 class Region(models.Model):
-    name = models.CharField(max_length=20)
+    name = models.CharField(max_length=30, unique=True)
 
 class District(models.Model):
+    EMPTY_FIELD = 'empty field'
```
[the remark this answers is not in the document]
Comment (Iva-khar) [reply ratu/models/ratu_models.py:10]: it's CharField fild. I may to do it as "number" bat why?

File: ratu/models/ratu_models.py (hunk 1 above)
Comment (Iva-khar) [reply ratu/models/ratu_models.py:15]: same

File: ratu/models/ratu_models.py (hunk 2 above)
Comment (Iva-khar) [reply ratu/models/ratu_models.py:21]: same

File: ratu/models/ruo_models.py (hunk 4 above)
Comment (Iva-khar) [reply ratu/models/ruo_models.py:13]: needs to discuss

File: ratu/models/rfop_models.py (hunk 3 above)
Comment (Iva-khar) [reply ratu/models/rfop_models.py:10]: +

File: ratu/services/main.py (hunk 5 above)
Comment (Iva-khar) [reply ratu/services/main.py:7]: +

File: ratu/services/main.py (hunk 6 above)
Comment (Iva-khar) [reply ratu/services/main.py:10]: +

File: ratu/services/main.py (hunk 7 above)
Comment (Iva-khar) [reply ratu/services/main.py:13]: +

File: ratu/services/main.py (hunk 8 above)
Comment (Iva-khar) [reply ratu/services/main.py:75]: +

File: ratu/models/ratu_models.py (hunk 11 above)
[the remark this answers is not in the document]
Comment (Iva-khar) [reply ratu/models/ratu_models.py:10]: done

File: ratu/models/ratu_models.py (hunk 1 above)
Comment (Iva-khar) [reply ratu/models/ratu_models.py:15]: done

File: ratu/models/ratu_models.py (hunk 2 above)
Comment (Iva-khar) [reply ratu/models/ratu_models.py:21]: done

File: ratu/services/main.py (hunk 5 above)
Comment (Iva-khar) [reply ratu/services/main.py:7]: done (deleted)

File: ratu/services/main.py (hunk 6 above)
Comment (Iva-khar) [reply ratu/services/main.py:10]: done

File: ratu/services/main.py (hunk 7 above)
Comment (Iva-khar) [reply ratu/services/main.py:13]: done

File: ratu/services/main.py (hunk 8 above)
Comment (Iva-khar) [reply ratu/services/main.py:75]: done

File: ratu/services/ratu_service.py (hunk 9 above)
Comment (Iva-khar) [reply ratu/services/ratu_service.py:49]: done

File: ratu/services/ratu_service.py (hunk 10 above)
Comment (Iva-khar) [reply ratu/services/ratu_service.py:66]: done

File: data_converter/settings.py [hunk 12]
```diff
@@ -11,6 +11,7 @@
 """
 
 import os
+import ratu.config as config
```
Comment (criticus) [inline data_converter/settings.py:14]: move this to upper level

File: ratu/models/__init__.py [hunk 13]
```diff
@@ -0,0 +1,3 @@
+from .ratu_models import Region, District, City, Citydistrict, Street
+from .rfop_models import Rfop
+from .ruo_models import Founders, Ruo, State_Ruo
```
Comment (criticus) [inline ratu/models/__init__.py:3]: follow naming convention

File: ratu/models/rfop_models.py [hunk 14]
```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
```
Comment (criticus) [inline ratu/models/rfop_models.py:2]: please remove unused

File: ratu/models/rfop_models.py [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+from django.db import models
+
+
+class State_Rfop(models.Model):
```
Comment (criticus) [inline ratu/models/rfop_models.py:6]: follow naming convention

File: ratu/models/rfop_models.py [hunk 16]
```diff
@@ -0,0 +1,16 @@
...
+
+    EMPTY_FIELD = 'empty field'
+    name = models.CharField(max_length=100, unique=True, null=True)
+
+class Rfop(models.Model):
+    
+    state = models.ForeignKey(State_Rfop, on_delete=models.CASCADE)
+    fullname = models.CharField(max_length=100, null=True)
+    address = models.CharField(max_length=300, null=True)
+    kved = models.CharField(max_length=250, null=True)
```
Comment (criticus) [inline ratu/models/rfop_models.py:16]: move Kved to separate table

File: ratu/models/ruo_models.py [hunk 17]
```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
```
Comment (criticus) [inline ratu/models/ruo_models.py:2]: remove unused

File: ratu/models/ruo_models.py [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+
+class Ruo(models.Model):
+
+    state = models.ForeignKey(State_Ruo, on_delete=models.CASCADE)
+    name = models.CharField(max_length=500, null=True)
+    short_name = models.CharField(max_length=500, null=True)
+    edrpou = models.CharField(max_length=50, null=True)
+    address = models.CharField(max_length=500, null=True)
+    boss = models.CharField(max_length=250, null=True)
+    kved = models.CharField(max_length=500, null=True)
```
Comment (criticus) [inline ratu/models/ruo_models.py:18]: ForeignKey?

File: ratu/services/main.py [hunk 19]
```diff
@@ -0,0 +1,72 @@
...
+
+class Converter:
+    
+    def __init__(self):
+        return 
+
+    def unzip_file(self):
+        # getting zip file  from FILE_URL & extracting to LOCAL_FOLDER
+        try:
+            r = requests.get(self.FILE_URL)
```
Comment (criticus) [inline ratu/services/main.py:17]: define const in class

File: data_converter/settings.py (hunk 12 above)
Comment (Iva-khar) [reply data_converter/settings.py:14]: done

File: ratu/models/__init__.py (hunk 13 above)
Comment (Iva-khar) [reply ratu/models/__init__.py:3]: done

File: ratu/models/rfop_models.py (hunk 14 above)
Comment (Iva-khar) [reply ratu/models/rfop_models.py:2]: done

File: ratu/models/rfop_models.py (hunk 15 above)
Comment (Iva-khar) [reply ratu/models/rfop_models.py:6]: done

File: ratu/models/ruo_models.py (hunk 17 above)
Comment (Iva-khar) [reply ratu/models/ruo_models.py:2]: done

File: ratu/services/main.py (hunk 19 above)
Comment (Iva-khar) [reply ratu/services/main.py:17]: done

**Pull request merged**